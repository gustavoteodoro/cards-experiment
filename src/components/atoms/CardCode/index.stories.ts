import CardCode from './index.vue';

export default { title: 'Atoms/CardCode' };

export const Default = () => ({
  components: { CardCode },
  template: '<card-code value="KING" suit="CLUBS" />'
});

export const suitDiamonds = () => ({
  components: { CardCode },
  template: '<card-code value="2" suit="DIAMONDS" />'
});

export const suitHearts = () => ({
  components: { CardCode },
  template: '<card-code value="ACES" suit="HEARTS" />'
});

export const suitSpades = () => ({
  components: { CardCode },
  template: '<card-code value="8" suit="SPADES" />'
});

export const suitClubs = () => ({
  components: { CardCode },
  template: '<card-code value="KING" suit="CLUBS" />'
});

export const Inverted = () => ({
  components: { CardCode },
  template: '<card-code value="8" suit="HEARTS" inverted />'
});
