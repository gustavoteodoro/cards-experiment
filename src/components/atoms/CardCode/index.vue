<template>
  <div :style="containerStyles">
    <span>{{ `${value.charAt(0)}` }}</span>
    <img :alt="suit" v-bind:src="require(`../../../assets/${suit.toLowerCase()}.png`)">
  </div>
</template>

<script>
export default {
  name: 'CardCode',
  props: {
    value: String,
    suit: String,
    inverted: Boolean,
  },
  computed: {
    containerStyles() {
      return {
        transform: `rotate(${this.inverted ? '180' : '0'}deg)`,
      };
    },
  },
};
</script>

<style scoped>
span {
  font-size: 30px;
  margin-right: 7px;
  display: inline-block;
  font-weight: bold;
}
img {
  width: 25px;
  margin-bottom: -7px;
}
</style>
