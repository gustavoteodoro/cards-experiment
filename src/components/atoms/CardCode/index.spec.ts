import { mount } from "@vue/test-utils";
import CardCode from "./index.vue"

describe("CardCode", () => {
  it('renders correctly King', () => {
    const wrapper = mount(CardCode , {
      propsData: {
        value: "KING",
      }
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
