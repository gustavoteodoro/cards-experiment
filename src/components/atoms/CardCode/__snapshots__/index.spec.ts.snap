// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardCode renders correctly 2 Spades 1`] = `
<div
  style="transform: rotate(0deg);"
>
  <span>
    2
  </span>
   
  <img />
</div>
`;

exports[`CardCode renders correctly King Diamonds 1`] = `
<div
  style="transform: rotate(0deg);"
>
  <span>
    K
  </span>
   
  <img />
</div>
`;
